ckanext/vocabulary/blueprints/vocabulary.py

```diff
@@ -183,19 +191,38 @@ def new_tags(id):
                 tag_dict = {}
                 tag_dict['name'] = tag[0].strip()
                 tag_dict['name_translated-en'] = tag[0].strip()
-                tag_dict['name_translated-ar'] = tag[1]
+                tag_dict['name_translated-ar'] = tag[1].strip()
                 tag_dict['vocabulary_id'] = id
 
+                non_free_tag_key = list(non_free_tags.keys())
+                en_tag = tag[0].lower().strip()
+                ar_tag = tag[1].lower().strip()
                 try:
-                    if tag[0].strip() in free_tags or tag[0].strip() in non_free_tags:
+                    if tag[0].lower().strip() in free_tags:
+                        error_tags +=1
+                        raise logic.ValidationError(
+                            error_dict= {'name_translated-en': [f'Tag: "{tag[0].strip()}"  already exist in Free Tags']}
+                        )
+                    elif en_tag in non_free_tag_key:
+                        error_tags +=1
+                        tagname = non_free_tags[en_tag]
+                        raise logic.ValidationError(
+                            error_dict= {'name_translated-en': [f'Tag: "{tag[0].strip()}"  already exist in Tag: "{tagname}"']}
```

Comment:
@steveoni 

Is this correct?
"Tag: <tagname>  already exist in Tag: <tagname>"

Or should this also mention the category in which the tag is already present like in line 219?
Eg. "Tag: <tagname>  already exist in Category <vocab_name> as <tagname>"